api/src/main/java/org/pmiops/workbench/cdrselector/CdrSelectorServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package org.pmiops.workbench.cdrselector;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.pmiops.workbench.cohortreview.CohortReviewService;
+import org.pmiops.workbench.conceptset.ConceptSetService;
+import org.pmiops.workbench.db.dao.DataSetDao;
+import org.pmiops.workbench.db.model.DbCohort;
+import org.pmiops.workbench.db.model.DbCohortReview;
+import org.pmiops.workbench.db.model.DbConceptSet;
+import org.pmiops.workbench.db.model.DbDataset;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.model.WorkspaceResource;
+import org.pmiops.workbench.utils.WorkspaceMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CdrSelectorServiceImpl implements CdrSelectorService {
+  private final CohortReviewService cohortReviewService;
+  private final ConceptSetService conceptSetService;
+  private final DataSetDao dataSetDao;
+  private final WorkspaceMapper workspaceMapper;
+
+  @Autowired
+  public CdrSelectorServiceImpl(
+      CohortReviewService cohortReviewService,
+      ConceptSetService conceptSetService,
+      DataSetDao dataSetDao,
+      WorkspaceMapper workspaceMapper) {
+    this.cohortReviewService = cohortReviewService;
+    this.conceptSetService = conceptSetService;
+    this.dataSetDao = dataSetDao;
+    this.workspaceMapper = workspaceMapper;
+  }
+
+  @Override
+  public List<WorkspaceResource> getCdrSelectorsInWorkspace(
```

Comment:
That's not something you can just casually tack on. It would involve such a fundamental redesign of the current model that it's not useful to speculate about how we might name that.

In frontend work, there's often a pattern of combining many things that are commonly request together into one endpoint to reduce rendering latency. The common thread connecting these things is that there is a page in the UI that is trying to render all of them at once. I would not overfit to that (alternatively: be transparent about overfitting) or you run the risk of being too inflexible when the product requirements inevitably evolve, e.g.: "going forward, lets split cohorts out into it's own tab" or "lets add notebooks here too".

Anyways, that's fine - I'll wait for the document.

Reply:
Hmmm... Let me actually give that a shot. I think it may be possible to have it take an array of Resource types. I will try to add that. If it is a lot of extra work, it may be worth splitting out until after discussion, but that sounds better. I pushed back against it a while ago, but thinking more, it shouldn't blow up too much scope if I just make the other types unsupported for now.